crates/cubecl-matmul/src/base.rs

```diff
@@ -216,6 +262,49 @@ impl<'a, R: Runtime> MatmulInputHandleRef<'a, R> {
             MatmulInputHandleRef::Quantized { shape, .. } => shape,
         }
     }
+
+    pub fn into_contiguous<E: Numeric>(
+        &self,
+        client: &ComputeClient<R::Server>,
+    ) -> MatmulInputHandle<R, E> {
+        match self {
+            MatmulInputHandleRef::Normal(data) => {
+                MatmulInputHandle::Normal(into_contiguous_pitched::<R, E>(client, data))
+            }
+            MatmulInputHandleRef::Quantized {
+                data,
+                scale,
+                shape,
+                scheme,
+            } => {
+                let data = match scheme.store {
+                    // e2m1 has native packing so also needs to be re-packed
+                    QuantStore::Native if scheme.value == QuantValue::E2M1 => {
+                        let data = into_contiguous_packed::<R, u8>(client, data, shape, 2);
```

Comment:
Why 2? Is it always the case, if so we should add a comment.

Reply:
Actually we need to assert the packed type (`u8`) so it should be specific to e2m1.